pkg/configuration/configuration.go

```diff
@@ -0,0 +1,94 @@
+// Package configuration is in charge of the validation and extraction of all
+// the configuration details from a configuration file or environment variables.
+package configuration
+
+import (
+	"os"
+	"strings"
+
+	errs "github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+// prefixes
+const (
+	// MemberEnvPrefix will be used for member environment variable name prefixing.
+	MemberEnvPrefix = "MEMBER_OPERATOR"
+)
+
+// Configuration constants
+const (
+	// ToolchainConfigMapName specifies a name of a ConfigMap that keeps toolchain configuration
+	ToolchainConfigMapName = "toolchain-saas-config"
+
+	// IdentityProvider specifies an identity provider (IdP) for newly created users
+	IdentityProvider = "identity.provider"
+
+	// DefaultIdentityProvider the default value used for the identity provider (IdP) for newly created users
+	DefaultIdentityProvider = "rhd"
+)
+
+// Registry encapsulates the Viper configuration registry which stores the
+// configuration data in-memory.
+type Registry struct {
```

Comment:
I'm the one who introduced this `Registry` type name a while ago (on OSIO) but I believe that `configuration.Configuration` would be more explicit, honestly (despite the repetition) 😬

Reply:
Haha, agree. I renamed it from `Registry` to `Config` so we can keep it short.